packages/foundation-models/src/azure-openai/azure-openai-chat-completion-stream.ts

```diff
@@ -0,0 +1,129 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import { Stream } from './azure-openai-stream.js';
+import { AzureOpenAiChatCompletionStreamChunkResponse } from './azure-openai-chat-completion-stream-chunk-response.js';
+import type { HttpResponse } from '@sap-cloud-sdk/http-client';
+import type { AzureOpenAiChatCompletionStreamResponse } from './azure-openai-chat-completion-stream-response.js';
+
+const logger = createLogger({
+  package: 'foundation-models',
+  messageContext: 'azure-openai-chat-completion-stream'
+});
+
+/**
+ * Chat completion stream containing post-processing functions.
+ */
+// TODO: The Item type `any` should actually be the type of the stream chunk response.
```

Comment:
In what way are we dependent on the spec here?

The way we solved this in Java is that we only return Stream<String> for the simple variant. This has the downside that the token usage, finish reason etc isn't available, but it's much simpler, both in using and for our coding.

While the capability to get both, token usage and a stream of string , is nice, I'd argue that knowing the stream type is more important

Reply:
`chunk` is now typed.